File: Mesh4j/trunk/FeedSync/src/com/feed/sync/model/Feed.java
package com.feed.sync.model;

import com.feed.sync.validations.Guard;

public class Feed {

	// MODEL VARIABLES
	private String title;
	private String description;
	private String link;
	private Object payload;
	private Sharing sharing = new Sharing();
	
	// BUSINESS METHODS
	
	public Feed(String title, String linkUrl, String description) 
	{
		this(title, linkUrl, description, null);
	}

	public Feed(String title, String linkUrl, String description, Object payload)
	{
		Guard.argumentNotNullOrEmptyString(title, "title");
		Guard.argumentNotNullOrEmptyString(linkUrl, "linkUrl");
	
		if (payload == null)
		{
			//TODO (?): this.payload = new XmlDocument().CreateElement("payload");
			this.payload = payload;
		}
		else
		{
			this.payload = payload;
		}
	
		this.title = title;
		this.link = linkUrl;
		this.description = description;
	}

	public String getTitle() {
		return title;
	}

	public String getDescription() {
		return description;
	}

	public String getLink() {
		return link;
	}

	public Object getPayload() {
		return payload;
	}

	public void setPayload(Object payload) {
		this.payload = payload;
	}

	public Sharing getSharing() {
		return sharing;
	}

	public void setSharing(Sharing sharing) {
		this.sharing = sharing;
	}

}

File: Mesh4n/trunk/Main/Source/Model/Feed.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Xml;

namespace Mesh4n
{
	[Serializable]
	public class Feed
	{
		private Sharing sharing = new Sharing();

		private string title;
		private string description;
		private string link;
		private XmlElement payload;

		// TODO: unpublished. 2.6 and 4.1. Add Unpublish(Item) ?

		public Feed() 
		{
		}

		public Feed(string title, string linkUrl, string description) 
			: this(title, linkUrl, description, null)
		{
		}

		public Feed(string title, string linkUrl, string description, XmlElement payload)
		{
			if (payload == null)
			{
				this.payload = new XmlDocument().CreateElement("payload");
			}
			else
			{
				this.payload = payload;
			}

			this.title = title;
			this.link = linkUrl;
			this.description = description;
		}

		public Sharing Sharing
		{
			get { return sharing; }
			set { sharing = value; }
		}

		public string Description
		{
			get { return description; }
		}

		public string Link
		{
			get { return link; }
		}

		public string Title
		{
			get { return title; }
		}

		public XmlElement Payload
		{
			get { return payload; }
			set { payload = value; }
		}
	}
}
